src/modules/currency/currencies.ts

```diff
@@ -0,0 +1,170 @@
+import BigNumber from 'bignumber.js';
+import memoize from 'memoize-one';
+
+interface ResolvedNumberFormatOptions {
+  numberingSystem: string;
+  style: string;
+  currency?: string;
+  currencyDisplay?: string;
+  minimumIntegerDigits: number;
+  minimumFractionDigits: number;
+  maximumFractionDigits: number;
+  minimumSignificantDigits?: number;
+  maximumSignificantDigits?: number;
+  useGrouping: boolean;
+}
+
+export enum CurrencyType {
+  Fiat,
+  Crypto,
+}
+
+export interface CurrencyConfig {
+  name: string;
+  code: string;
+  symbol: string;
+  locale: string;
```

Comment:
Do we need locale? We're not using it

Reply:
No, left this for the future, but let's remove it

Reply:
Locale shouldn't be really tied to currency I believe)